scripts/system_performance/run_statistics.py

```diff
@@ -0,0 +1,292 @@
+            # Want to look in current cycle
+            run_performance_count = self.run_fail_count(start, end)
+
+            in_cycle = None
+            return in_cycle
+
+        # checking if current date is between cycles (# "index+1" to look at next rows start date)
+        if index < self.data_frame.tail(1).index.item():
+            if self.current_date > end < self.cycle_search([self.start_col], index+1):
+                # want to look at previous cycle
+                dates = self.cycle_search([self.start_col, self.end_col],index+1)
+                start, end = dates[0], dates[1]
+                run_performance_count = self.run_fail_count(start, end)
+
+                in_cycle = None
+                return in_cycle
+            # Edge case for current date not in a cycle or between cycles (covers maintenance days when added)
+            else:
+                in_cycle = None
+                return in_cycle
+        # Looking at last row in data frame. Current date can't be checked any further back in time
+        else:
+            in_cycle = None
+            return in_cycle
+
+    def create_method_mappings(self):
+        """A dictionary to map on input, user specified methods to return to equivalent method to be called"""
+        method_mappings = {'missing_run_numbers_report': self.missing_run_numbers_report,
+                           'execution_time': self.execution_time,
+                           'execution_time_average': self.execution_time_average,
+                           'run_frequency': self.run_frequency,
+                           'run_frequency_average': self.run_frequency_average
+                           }
+        return method_mappings
+
+    @staticmethod
+    def list_all_instruments():
+        """Returns a list of all currently active instruments"""
+        instrument_list = DatabaseMonitorChecks.list_instruments()
+        return instrument_list
+
+    def run_every_instruments(self, instrument_dict, method_name):
+        """Returns all existing instruments in a dictionary for a given method as:
+        {instrument_name : [[method output row 1]],[[method output row 2] etc] ...}"""
+        for sublist in self.list_all_instruments():
+            try:
+                if method_name in self.create_method_mappings():
+                    instrument_dict[sublist[1]] = self.create_method_mappings()[method_name](sublist[0])
+            except KeyError:
+                raise ValueError('Invalid Input - method {} does not exist'.format(method_name))
+        return instrument_dict
+
+    def input_verification(self, instrument_input, method_name):
+        """Checks that the instrument's in method input exist and then calls
+        and returns the methods used as input for each instrument placing in
+        a dictionary as a nested list for each instrument as:
+         {instrument_name : [[method output row 1]],[[method output row 2] etc.] ...}"""
```

Comment:
This needs some rewording as it's very difficult to follow I would suggest adding doxygen style comments to state the params and their types and simplifying the function description:
```
Run methods for each instrument name
:param instrument_input: [(str), (str), ...] - Represents the instruments to be used in queries
:param method_name: (str) -  The name of the methods to run
:return: A dictionary of instrument name (key) and list of method output (value)
```